papi/__init__.py

```diff
@@ -18,7 +18,8 @@ class Papi:
     edges: List[Edge]
     counters: Counter
 
-    def __init__(self, config: Config):
+    def __init__(self, name: str, config: Config):
+        self.name = name
```

Comment:
Maybe

```suggestion
    def __init__(self, config: Config, *, name: str = 'Pipeline'):
        self.name = str(name)
```
?

The name is not strictly necessary (in e.g. API use).

Reply:
I'm ok with this